src/System.Net.Primitives/src/System/Net/IPAddressParser.cs

```diff
@@ -54,24 +84,79 @@ internal static unsafe string IPv4AddressToString(uint address)
         {
             const int MaxLength = 15;
             char* addressString = stackalloc char[MaxLength];
-            int offset = MaxLength;
 
-            FormatIPv4AddressNumber((int)((address >> 24) & 0xFF), addressString, ref offset);
-            addressString[--offset] = '.';
-            FormatIPv4AddressNumber((int)((address >> 16) & 0xFF), addressString, ref offset);
-            addressString[--offset] = '.';
-            FormatIPv4AddressNumber((int)((address >> 8) & 0xFF), addressString, ref offset);
-            addressString[--offset] = '.';
+            int charsWritten = IPv4AddressToStringHelper(address, addressString);
+
+            return new string(addressString, 0, charsWritten);
+        }
+
+        internal static unsafe bool IPv4AddressToString(uint address, Span<char> formatted, out int charsWritten)
+        {
+            const int MaxLength = 15;
+
+            if (formatted.Length < MaxLength)
+            {
+                charsWritten = 0;
+                return false;
+            }
```

Comment:
Seems like this method should be made a private helper, and then another internal wrapper for it should be added that does this check.  It's not needed when called from the string-returning version, only for the span one.